packages/esm-service-queues-app/src/queue-services/queue-service-form.workspace.tsx

```diff
@@ -17,86 +21,91 @@ import { type DefaultWorkspaceProps, restBaseUrl, showSnackbar } from '@openmrs/
   const { queueConcepts } = useServiceConcepts();
   const { queueLocations } = useQueueLocations();
-  const [queueName, setQueueName] = useState('');
-  const [queueConcept, setQueueConcept] = useState('');
-  const [isMissingName, setMissingName] = useState(false);
-  const [isMissingQueue, setMissingQueue] = useState(false);
-  const [isMissingLocation, setMissingLocation] = useState(false);
-  const [userLocation, setUserLocation] = useState('');
-
-  const createQueue = useCallback(
-    (event) => {
-      event.preventDefault();
-
-      if (!queueName) {
-        setMissingName(true);
-        return;
-      }
-      if (!queueConcept) {
-        setMissingQueue(true);
-        return;
-      }
-      if (!userLocation) {
-        setMissingLocation(true);
-        return;
-      }
 
-      setMissingName(false);
-      setMissingQueue(false);
-      setMissingLocation(false);
-
-      saveQueue(queueName, queueConcept, queueName, userLocation).then(
-        ({ status }) => {
-          if (status === 201) {
-            showSnackbar({
-              title: t('addQueue', 'Add queue'),
-              kind: 'success',
-              subtitle: t('queueAddedSuccessfully', 'Queue added successfully'),
-            });
-            closeWorkspace();
-            mutate(`${restBaseUrl}/queue?${userLocation}`);
-            mutate(`${restBaseUrl}/queue?location=${userLocation}`);
-          }
-        },
-        (error) => {
-          showSnackbar({
-            title: t('errorAddingQueue', 'Error adding queue'),
-            kind: 'error',
-            isLowContrast: false,
-            subtitle: error?.message,
-          });
-        },
-      );
+  const {
+    control,
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<QueueServiceFormData>({
+    resolver: zodResolver(QueueServiceSchema),
+    defaultValues: {
+      queueName: '',
+      queueConcept: '',
+      userLocation: '',
     },
-    [queueName, queueConcept, userLocation, t, closeWorkspace],
-  );
+  });
+
+  const createQueue = (data: QueueServiceFormData) => {
+    saveQueue(data.queueName, data.queueConcept, data.queueName, data.userLocation)
+      .then(() => {
+        showSnackbar({
+          title: t('addQueue', 'Add queue'),
+          kind: 'success',
+          subtitle: t('queueAddedSuccessfully', 'Queue added successfully'),
+        });
+        closeWorkspace();
+        mutate(`${restBaseUrl}/queue?${data.userLocation}`);
+        mutate(`${restBaseUrl}/queue?location=${data.userLocation}`);
+      })
+      .catch((error) => {
+        showSnackbar({
+          title: t('errorAddingQueue', 'Error adding queue'),
+          kind: 'error',
+          isLowContrast: false,
+          subtitle: error?.message,
+        });
+      });
+  };
 
   return (
-    <Form onSubmit={createQueue} className={styles.form}>
+    <Form onSubmit={handleSubmit(createQueue)} className={styles.form}>
       <Stack gap={4} className={styles.grid}>
```

Comment:
```suggestion
      <Stack gap={5} className={styles.grid}>
```

The extra vertical space looks cleaner to me.